bob/build_nix.go

```diff
@@ -3,14 +3,30 @@ package bob
 import (
 	"fmt"
 
+	"github.com/benchkram/errz"
+
 	"github.com/benchkram/bob/bob/bobfile"
+	cache2 "github.com/benchkram/bob/pkg/cache"
 	"github.com/benchkram/bob/pkg/nix"
 	"github.com/benchkram/bob/pkg/usererror"
-	"github.com/benchkram/errz"
 )
 
+// Nix builder acts as a wrapper for github.com/benchkram/bob/pkg/nix package
+// and is used for building tasks dependencies
+type Nix struct {
+	// cache allows caching the dependency to store path
+	cache cache2.Cache
+}
+
+// NewNix instantiates a new Nix builder instance
+func NewNix(cache cache2.Cache) *Nix {
```

Comment:
Make sure to initialize it by default with the correct cache.  For testing pass one using the optional pattern WithCache(cache).